### PR TITLE
Added a UDF to return a friendly Windows name (as a string) 

### PR DESCRIPTION
**Pull request by perrymcmanis144**
pursuant to: https://mozilla.slack.com/archives/C4D5ZA91B/p1665686262468709

This uses os, os_version and windows_build_number

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql/mozfun/norm/windows_version_info/udf.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
+CREATE OR REPLACE FUNCTION norm.windows_version_info(os STRING, os_version STRING, windows_build_number INT64)
+RETURNS STRING AS (
+  CASE
+    WHEN os NOT IN ('Windows_NT', 'Windows_95', 'Windows_98') THEN NULL
+    WHEN os = 'Windows_95' THEN 'Windows 95'
+    WHEN os = 'Windows_98' THEN 'Windows 98'
+    WHEN os_version = '6.1' THEN 'Windows 7'
```
Comment (fbertsch) [inline sql/mozfun/norm/windows_version_info/udf.sql:7]: nit: can these be reordered?

File: sql/mozfun/norm/windows_version_info/udf.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    WHEN os_version = '10.0' AND windows_build_number < 22000 THEN 'Windows 10'
+    WHEN os_version = '4.0' AND os = 'Windows_NT' THEN 'Windows NT 4.0'
+    WHEN os_version = '5.0' THEN 'Windows 2000'
+    ELSE NULL
+  END
+);
+
+-- Tests
+SELECT
+  assert.equals('Windows 10', norm.windows_version_info('Windows_NT', '10.0', 19043)),
```
Comment (fbertsch) [inline sql/mozfun/norm/windows_version_info/udf.sql:21]: We should test for more scenarios than this

**Review by fbertsch — CHANGES_REQUESTED** [review]
Thank you for writing this, Perry! We should add this to clients_daily as soon as it's ready.

I'm seeing some missing data by this query: https://sql.telemetry.mozilla.org/queries/88103/source, e.g. os_version `6.3` would return `NULL` by this UDF.

Would you be willing to add a link to a query like that in the tests, and make sure each of those scenarios are covered?

File: sql/mozfun/norm/windows_version_info/udf.sql (hunk 2 above)
Comment (perrymcmanis144) [reply sql/mozfun/norm/windows_version_info/udf.sql:28]: more test scenarios added, I think the should cover more corner cases now (also, I reordered to hopefully make organization better)

File: sql/mozfun/norm/windows_version_info/udf.sql (hunk 1 above)
Comment (perrymcmanis144) [reply sql/mozfun/norm/windows_version_info/udf.sql:11]: Also handled 6.3

File: sql/mozfun/norm/windows_version_info/udf.sql [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  norm.windows_version_info( os STRING,
+    os_version STRING,
+    windows_build_number INT64 )
+  RETURNS STRING AS (
+    CASE
+      WHEN os NOT IN ('Windows_NT', 'Windows_95', 'Windows_98') THEN NULL
+      WHEN os = 'Windows_95' THEN 'Windows 95'
+      WHEN os = 'Windows_98' THEN 'Windows 98'
+      WHEN os_version = '4.0'
+    AND os = 'Windows_NT' THEN 'Windows NT 4.0'
```
Comment (fbertsch) [inline sql/mozfun/norm/windows_version_info/udf.sql:11]: SQL formatter, what did you do 😭

File: sql/mozfun/norm/windows_version_info/udf.sql [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      WHEN os_version = '5.0' THEN 'Windows 2000'
+      WHEN os_version = '5.1'
+    OR os_version = '5.2' THEN 'Windows XP'
+      WHEN os_version = '6.0' THEN 'Windows Vista'
+      WHEN os_version = '6.1' THEN 'Windows 7'
+      WHEN os_version = '6.2' THEN 'Windows 8'
+      WHEN os_version = '6.3' THEN 'Windows 8.1'
```
Comment (fbertsch) [inline sql/mozfun/norm/windows_version_info/udf.sql:18]: Do we want to differentiate these? Or lump them into `Windows 8`?

**Review by fbertsch — APPROVED** [review]
r+wc, thank you for doing this Perry 🙏

File: sql/mozfun/norm/windows_version_info/udf.sql (hunk 3 above)
Comment (perrymcmanis144) [reply sql/mozfun/norm/windows_version_info/udf.sql:11]: not sure, this is actually the google one because I cannot get ./bqetl format --check to run on my m1. I would very much like to take you up on your offer that once you're satisfied with the UDF itself that you could recommit with the formatter run.

File: sql/mozfun/norm/windows_version_info/udf.sql (hunk 4 above)
Comment (perrymcmanis144) [reply sql/mozfun/norm/windows_version_info/udf.sql:18]: I think for analysis it's fine to have these separated out, if folks find that grouping them becomes something that happens more often then not then perhaps we update it.

In the case of weirdness, it could potentially make it easier to track something down. And I guess MSFT thinks they are different enough to have them separated out.
